coroner-client/src/main/java/backtraceio/coroner/query/CoronerQueryBuilder.java

```diff
@@ -0,0 +1,44 @@
+package backtraceio.coroner.query;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class CoronerQueryBuilder {
+    private final String FOLD_HEAD = "head";
+    private final int OFFSET = 0;
+    private final int LIMIT = 1;
+
+    public String buildRxIdGroup(final String filters, final List<String> headFolds) {
+        return this.build(CoronerQueryFields.RXID, filters, headFolds);
+    }
+
+    private String build(final String groupName, final String filters, final List<String> headFolds) {
+        final String folds = joinHeadFolds(headFolds);
+
+        return "{" +
```

Comment:
why you building a json manually and not via library. What would happen if you get a character that breaks the JSON ?

Reply:
Let's keep it as it is because this query syntax is quite problematic like e.g. arrays of arrays with index dependent elements (check snippet below). We will use it in recent future only in integration tests so I don't want to spend more time on that. Of course you're right that it will be good to have it in the future.

```
    "guid": [
                [
                    "equal",
                    "2e55f002-32c6-56e4-9210-b54ac25be583"
                ]
            ],
```

Reply:
Next time, can you please don't resolve comments that don't address the resolution to the question? On Github it's super hard to find out what did we change and what we didn't - later I'm not sure if we finished something or not. 

The example above won't be serialized to `string[][]`? We know where can we find information. 


if you want to stay with the custom JSON, how you're dealing with escapes or non cultural characters - use emoji for example. 

This is a matter of time when it will break because of something super trivial

Reply:
Hi Konrad, I've decided to address your issue and now query JSON is build using library instead of string concatenation.